"Error initializing contentScript" because the storage is empty

I've been seeing this error for a while on https://app.pixiebrix.com/accounts/profile/ , it probably happens on fresh installs because `rawOptions` is `undefined`


```
contentScript.js:203921 Error initializing contentScript SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at loadOptions (contentScript.js:191669)
    at async loadExtensions (contentScript.js:182829)
    at async loadExtensionsOnce (contentScript.js:182861)
    at async handleNavigate (contentScript.js:182899)
    at async init (contentScript.js:203918)
```

On this line:

https://github.com/pixiebrix/pixiebrix-extension/blob/b7151eb9246373977ed80d7ccfc9b628f8eb9a21/src/options/loader.ts#L53


The page seems to load correctly, but I don't see `contentScript ready in 129ms`

--- a/src/options/loader.ts
+++ b/src/options/loader.ts
@@ -15,12 +15,13 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-import { readStorage, setStorage } from "@/chrome";
 import { Metadata, ServiceDependency } from "@/core";
 import { Permissions } from "webextension-polyfill-ts";
 import { Primitive } from "type-fest";
+import { browser } from "webextension-polyfill-ts";
 
-const storageKey = "persist:extensionOptions";
+const STORAGE_KEY = "persist:extensionOptions";
+const INITIAL_STATE = JSON.stringify({});
 
 export interface ExtensionOptions {
   id: string;
@@ -43,15 +44,23 @@ type ExtensionOptionState = {
   extensions: Record<string, Record<string, ExtensionOptions>>;
 };
 
+type JSONString = string;
+type RawOptionsState = Record<string, JSONString>;
+
+async function getOptionsState(): Promise<RawOptionsState> {
+  // eslint-disable-next-line security/detect-object-injection -- constant storage key
+  const rawOptions = (await browser.storage.local.get(STORAGE_KEY))[
+    STORAGE_KEY
+  ];
+  return JSON.parse((rawOptions as string) ?? INITIAL_STATE);
+}
+
 /**
  * Read extension options from local storage
  */
 export async function loadOptions(): Promise<ExtensionOptionState> {
-  const rawOptions = await readStorage(storageKey);
-
-  // Not really sure why the next level down is also escaped JSON?
-  const base = JSON.parse(rawOptions as string);
-
+  const base = await getOptionsState();
+  // The redux persist layer persists the extensions value as as JSON-string
   return { extensions: JSON.parse(base.extensions) };
 }
 
@@ -59,11 +68,12 @@ export async function loadOptions(): Promise<ExtensionOptionState> {
  * Save extension options to local storage
  */
 export async function saveOptions(state: ExtensionOptionState): Promise<void> {
-  const rawOptions = await readStorage(storageKey);
-  const base = JSON.parse(rawOptions as string);
-  const { extensions } = state;
-  await setStorage(
-    storageKey,
-    JSON.stringify({ ...base, extensions: JSON.stringify(extensions) })
-  );
+  const base = await getOptionsState();
+  await browser.storage.local.set({
+    // The redux persist layer persists the extensions value as as JSON-string
+    [STORAGE_KEY]: JSON.stringify({
+      ...base,
+      extensions: JSON.stringify(state.extensions),
+    }),
+  });
 }

